go/apis/kube-mail/v1alpha1/emailpolicy_types.go

```diff
@@ -0,0 +1,56 @@
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type EmailPolicySMTPSink struct {
+	Server      v1.ObjectReference `json:"server"`
+	Credentials v1.ObjectReference `json:"credentials"`
```

Comment:
The use of `v1.ObjectReference` (at least in specs, where they're supposed to be filled in by a user) is [discouraged nowadays](https://pkg.go.dev/k8s.io/api/core/v1?utm_source=godoc#ObjectReference):

> New uses of this type are discouraged because of difficulty describing its usage when embedded in APIs. [...] **Instead of using this type, create a locally provided and used type that is well-focused on your reference.**